src/toast/ops/run_spt3g.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (c) 2021-2021 by the parties listed in the AUTHORS file.
+# All rights reserved.  Use of this source code is governed by
+# a BSD-style license that can be found in the LICENSE file.
+
+import traitlets
+
+import numpy as np
+
+from ..utils import Logger
+
+from ..traits import trait_docs, Int, Unicode, Bool, Instance, List
+
+from ..timing import function_timer
+
+from .operator import Operator
+
+from ..spt3g import available
+
+if available:
+    from spt3g import core as c3g
+    from ..spt3g import frame_emitter
+    from ..spt3g import frame_collector
+
+
+@trait_docs
+class RunSpt3g(Operator):
```

Comment:
When converting the observation data to frames, the export function already accepts a set of intervals (i.e. a view) to use for the frame boundaries:  https://github.com/hpc4cmb/toast/blob/24f6b743d19c38df727e47fec037bd3770716208/src/toast/spt3g/spt3g_export.py#L339
If that is specified, then only those intervals will be emitted as frames to be processed by the G3 pipeline run by this operator.  The default frame boundaries are set by the sample set boundaries in the observation distribution.

However, I just realized that using a set of intervals to emit frames with a subset of data would have an interesting consequence when re-converting these frames back into an observation.  In that situation, the output observation would only contain data from frames that passed through the G3 pipeline.  So all of the data not in the original intervals would be cut out.  

Not sure if there is a way around this- the G3 model for cutting data is to just drop those frames.  If we wanted to support this, we would have to pass a pre-existing observation to the `import_obs` class and have that class take each frame and try to locate it with the observation.  Then it would have to copy all the data in the frame into the corresponding observation shared / detdata fields at the correct sample offset.  And it would have to create any new data objects that were placed in the frames by the G3 pipeline.  It is a lot of extra logic...

For the short term, we could simply not use that option if we intend to return data back to toast observations.